sc/src/test/java/uk/ac/ebi/atlas/metadata/CellMetadataDaoIT.java

```diff
@@ -151,6 +167,19 @@ void getFieldValuesForNoFieldsReturnsEmpty() {
         return jdbcUtils.getPublicSingleCellExperimentAccessions()
                 .stream()
                 .filter(accession ->
-                        !accession.equalsIgnoreCase("E-GEOD-99058") && !accession.equalsIgnoreCase("E-ENAD-13"));
+                        !accession.equalsIgnoreCase("E-GEOD-99058")
+                                && !accession.equalsIgnoreCase("E-ENAD-13"));
+    }
+
+    private Stream<String> experimentsWithAdditionalAttributesProvider() {
+        return jdbcUtils.getPublicSingleCellExperimentAccessions()
+                .stream()
+                .filter(accession -> !idfParser.parse(accession).getMetadataFieldsOfInterest().isEmpty());
+    }
+
+    private Stream<String> experimentsWithoutAdditionalAttributesProvider() {
+        return jdbcUtils.getPublicSingleCellExperimentAccessions()
+                .stream()
+                .filter(accession -> idfParser.parse(accession).getMetadataFieldsOfInterest().isEmpty());
```

Comment:
Notice that for both choosing the experiments with or without additional metadatafields, and for checking the results in the tests, you’re following the same code path: `idfParser.parse(experimentAccession).getMetadataFieldsOfInterest()`.

In order to alleviate that I think it would be better to inject `DataFileHub` and scan the line with additional attributes, so that the experiment-picking logic is more isolated from the implementation of the test subject.